File: 프로그래머스/Level2/올바른 괄호.cpp
#include <string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<int> g;
    
    if(s[0] ==')') return false;
    
    for(int i=0; i<s.size(); i++){
        if(s[i] == '('){
            g.push(0);
        }
        else{
            if(!g.empty()){
                g.pop();
            }
        }
    }
    
    if(!g.empty()){
        answer = false;
    }

    return answer;
}

File: 프로그래머스/Level2/올바른 괄호.js
function solution(s){
    if(s[0] === ')') return false;
    
    let arr = [];
    
    for(let i=0; i<s.length; i++){
        if(s[i] === '(') arr.push(s[i]);
        else{
            if(arr.length === 0) return false;
            arr.pop();
        }
    }
    
    return arr.length === 0 ? true : false;
}
